app/controllers/api/v3/patients_controller.rb

```diff
@@ -42,15 +41,16 @@ def unscope_associations
   end
 
   def merge_if_valid(single_patient_params)
-    validator = Api::V3::PatientPayloadValidator.new(single_patient_params)
-    logger.debug "Patient had errors: #{validator.errors_hash}" if validator.invalid?
+    validator_params = single_patient_params.merge(request_user_id: current_user.id)
+    validator = Api::V3::PatientPayloadValidator.new(validator_params)
+
     if validator.invalid?
+      logger.debug "Patient had errors: #{validator.errors_hash}"
       NewRelic::Agent.increment_metric("Merge/Patient/schema_invalid")
       {errors_hash: validator.errors_hash}
     else
-      patients_params_with_metadata = single_patient_params.merge(metadata: metadata)
-      transformed_params = Api::V3::PatientTransformer.from_nested_request(patients_params_with_metadata)
-      patient = MergePatientService.new(transformed_params).merge
+      transformed_params = Api::V3::PatientTransformer.from_nested_request(single_patient_params)
+      patient = MergePatientService.new(transformed_params, request_metadata: request_metadata).merge
```

Comment:
We explicitly pass in the `request_metadata` to the service instead of merging it into the patient params. This avoids conflating the `request_user_id` and `registration_user_id` (same for facility) which behaves differently for new patients and patients being updated.

The other reason is that it also clearly segregates the omnipresent "metadata" from the actual parameters by treating them as a different entity: `request_metadata`.